include/mbgl/map/map.hpp

```diff
@@ -35,7 +36,8 @@ class Map : private util::noncopyable {
     explicit Map(RendererFrontend&,
                  MapObserver&,
                  const MapOptions&,
-                 const ResourceOptions&);
+                 const ResourceOptions&,
+                 const ClientOptions&);
```

Comment:
So maybe here we could use an optional argument. Or maybe we never use optional arguments in Native? If we change the constructor signature, I guess this means that this pull request introduces a breaking change of the core api. Do I understand that correctly? But maybe this has no consequences for users because they use the iOS and android bindings etc instead of the core?

Reply:
Yes, I updated all clients in the code to compile. Optionals are used in some places but are just harder to handle. Maybe I can use defaults, but I think this is such a core API that users would never need to use it. They'd just pass it to the `mbgl::Map`, and there I can maybe use a default.

Reply:
OK I guess the users don't see the core API directly. By the way, when we say core API we mean the stuff in include/mbgl, right?

Reply:
It's optional now for `mbgl::Map`, but not for map source classes.